iothub/device/src/Pipeline/MqttTransportHandler.cs

```diff
@@ -28,11 +28,13 @@ namespace Microsoft.Azure.Devices.Client.Transport
     internal sealed class MqttTransportHandler : TransportHandlerBase, IDisposable
     {
         private const int ProtocolGatewayPort = 8883;
+        private const int MessageQueueSize = 10;
```

Comment:
Setting the default queue size to 10. The client would always keep track of latest 10 messages.

Reply:
I suspect people are going to want to modify this value - why don't we expose this as a settable property?